From 0db2bdf58f2d614a1240e9592b77524dad73e598 Mon Sep 17 00:00:00 2001
From: Heitor Farhat <heitor.farhat@windriver.com>
Date: Tue, 2 Sep 2025 10:42:36 -0300
Subject: [PATCH] ceph-csi-rbd: add default fields to values.yaml

This patch adds fields and values necessary for the proper use
of the rbd chart in the system.

Signed-off-by: Heitor Farhat <heitor.farhat@windriver.com>
---
 charts/ceph-csi-rbd/values.yaml | 42 +++++++++++++++++++++++++++++++--
 1 file changed, 40 insertions(+), 2 deletions(-)

diff --git a/charts/ceph-csi-rbd/values.yaml b/charts/ceph-csi-rbd/values.yaml
index 2593b97..1fb94a9 100644
--- a/charts/ceph-csi-rbd/values.yaml
+++ b/charts/ceph-csi-rbd/values.yaml
@@ -337,10 +337,10 @@ topology:
   # - topology.kubernetes.io/region
   # - topology.kubernetes.io/zone

-# readAffinity:
+readAffinity:
 # Enable read affinity for RBD volumes. Recommended to
 # set to true if running kernel 5.8 or newer.
-# enabled: false
+enabled: false
 # Define which node labels to use as CRUSH location.
 # This should correspond to the values set in the CRUSH map.
 # NOTE: the value here serves as an example
@@ -526,6 +526,24 @@ storageClass:
   # mountOptions:
   #   - discard

+  # Ceph user name to access this pool
+  userId: kube
+  # K8 secret name with key for accessing the Ceph pool
+  userSecretName: ceph-secret-kube
+  # Pool replication
+  replication: 1
+  # Pool min replication
+  min_replication: 1
+  # Pool crush rule name
+  crush_rule_name: storage_tier_ruleset
+  # Pool chunk size / PG_NUM
+  chunk_size: 8
+  # Additional namespace to allow storage class access (other than where
+  # installed)
+  additionalNamespaces:
+  - default
+  - kube-public
+
 volumeSnapshotClass:
   # Specifies whether the VolumeSnapshotClass should be created
   create: false
@@ -650,6 +668,26 @@ externallyManagedConfigmap: false
 cephConfConfigMapName: ceph-config
 # Name of the configmap used for encryption kms configuration
 kmsConfigMapName: ceph-csi-encryption-kms-config
+
+#
+# Defaults for storage classes.
+#
+classdefaults:
+  # Define ip addresses of Ceph Monitors
+  monitors:
+    - 192.168.204.3:6789
+    - 192.168.204.150:6789
+    - 192.168.204.4:6789
+  # K8 secret name for the admin context
+  adminId: admin
+  adminSecretName: ceph-secret
+  # Default StorageClass name
+  storageClass: general
+
+storage_init:
+  image:
+    repository: docker.io/openstackhelm/ceph-config-helper
+    tag: ubuntu_jammy_18.2.2-1-20241210
 # Unique ID distinguishing this instance of Ceph CSI among other instances,
 # when sharing Ceph clusters across CSI instances for provisioning
 # instanceID: default
--
2.34.1
